SonarQube.TeamBuild.PreProcessor/SonarWebService.cs

```diff
@@ -222,17 +242,142 @@ public bool TryDownloadEmbeddedFile(string pluginKey, string embeddedFileName, s
+                return this.downloader.TryDownloadFileIfExists(url, targetFilePath);
+            }, url);
         }
 
         #endregion
 
         #region Private methods
 
+        private T DoLogExceptions<T>(Func<T> op, string url)
+        {
+            try
+            {
+                return op();
+            }
+            catch (Exception e)
+            {
+                this.logger.LogError("Failed to request and parse '{0}': {1}", url, e.Message);
+                throw;
+            }
+        }
+        private void DownloadServerVersion()
+        {
+            var ws = GetUrl("api/server/version");
+            serverVersion = DoLogExceptions(() =>
+            {
+                var contents = this.downloader.Download(ws);
+                int separator = contents.IndexOf('-');
+
+                if (separator >= 0)
+                {
+                    return new Version(contents.Substring(0, separator));
+                }
+                else
+                {
+                    return new Version(contents);
+                }
+            }, ws);
+        }
+
+        private IDictionary<string, string> GetPropertiesOld(string projectId)
+        {
+            string ws = GetUrl("/api/properties?resource={0}", projectId);
+            this.logger.LogDebug(Resources.MSG_FetchingProjectProperties, projectId, ws);
+            var result = DoLogExceptions(() =>
+            {
+                var contents = this.downloader.Download(ws);
+                var properties = JArray.Parse(contents);
+                return properties.ToDictionary(p => p["key"].ToString(), p => p["value"].ToString());
+            }, ws);
+            // http://jira.sonarsource.com/browse/SONAR-5891 or when C# plugin is not installed
+            if (!result.ContainsKey("sonar.cs.msbuild.testProjectPattern"))
+            {
+                result["sonar.cs.msbuild.testProjectPattern"] = SonarProperties.DefaultTestProjectPattern;
+            }
+
+            return result;
+        }
+
+        private IDictionary<string, string> GetProperties63(string projectId)
+        {
+            string ws = GetUrl("/api/settings/values?component={0}", projectId);
+            this.logger.LogDebug(Resources.MSG_FetchingProjectProperties, projectId, ws);
+            string contents = "";
+            bool success = DoLogExceptions(() => this.downloader.TryDownloadIfExists(ws, out contents), ws);
+
+            if (!success)
+            {
+                ws = GetUrl("/api/settings/values");
+                this.logger.LogDebug("No settings for project {0}. Getting global settings: {1}", projectId, ws);
+                contents = DoLogExceptions(() => this.downloader.Download(ws), ws);
+            }
+
+            return DoLogExceptions(() => parseSettingsResponse(contents), ws);
+        }
+
+        private Dictionary<string, string> parseSettingsResponse(string contents)
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Rename method 'parseSettingsResponse' to match camel case naming rules, consider using 'ParseSettingsResponse'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS100)